src/lean_spec/subspecs/forkchoice/store.py

```diff
@@ -261,23 +261,57 @@ def on_attestation(
             }
         )
 
-    def _validate_block_signatures(self, block: Block, signatures: BlockSignatures) -> bool:
+    def _validate_block_signatures(
+        self,
+        block: Block,
+        signatures: BlockSignatures,
+        proposer_attestation: Attestation,
+    ) -> bool:
         """
         Validate block signatures.
 
-        Temporary stub for aggregated signature validation.
-
         Args:
             block: Block to validate.
             signatures: Block signatures to validate.
+            proposer_attestation: Attestation from the block proposer.
 
         Returns:
-            True if all signatures are valid.
-
-        Note:
-            TODO: Integrate actual aggregated signature verification.
+            True if block signatures are valid, False otherwise.
         """
-        return all(Signature.is_valid(signature) for signature in signatures)
+        attestations = list(block.body.attestations)
+        attestations.append(proposer_attestation)
+
+        if len(signatures) != len(attestations):
+            return False
```

Comment:
Can we use assertions instead? Otherwise it'll be really hard for debugging especially when dealing with signatures

```suggestion
        assert len(signatures) == len(attestations), f"Signatures length ({len(signatures)}) does not match attestations length ({len(attestations)})"
```

Also applies to the rest of `return False`. There're way too many scenarios that `_validate_block_signatures()` can return False.